Automattic/wp-calypso
# Editor: Remove unused `ResumeEditing` component and last draft state

#### Changes proposed in this Pull Request

In the classic editor in Calypso, we used to keep the last draft in the state, and if the user leaves the editor, in the current session we displayed a "Continue Editing" button in the masterbar - see #4388 for more information.

However, when we removed the classic editor and replaced it with Gutenberg, we no longer store that last draft. We actually removed the last bit of code that was setting that last draft a while ago in #48352. 

Anyway, in that cleanup effort, we didn't remove `ResumeEditing` component that relied on that last draft being there. Without that last draft, that component would render nothing, so it can be removed. That component is also the last consumer of the last draft state, so that can be removed altogether too.

So this PR removes the `ResumeEditing` component, as well as the last draft state. 

I stumbled upon this when looking at addressing part of #58453 (the `ResumeEditing` component). This PR actually resolves that one item from #58453.

#### Testing instructions

* Verify the removed code is not in use.
* Verify Calypso builds without errors.
* Verify all tests still pass.
